qr-code/src/main/java/com/adyen/checkout/qrcode/FullQRCodeView.kt

```diff
@@ -0,0 +1,190 @@
+import android.os.Build
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.LinearLayout
+import androidx.annotation.RequiresPermission
+import androidx.annotation.StringRes
+import androidx.core.content.ContextCompat
+import androidx.core.view.isVisible
+import com.adyen.checkout.components.base.ComponentDelegate
+import com.adyen.checkout.components.base.ComponentParams
+import com.adyen.checkout.components.extensions.setLocalizedTextFromStyle
+import com.adyen.checkout.components.extensions.toast
+import com.adyen.checkout.components.image.LogoSize
+import com.adyen.checkout.components.image.load
+import com.adyen.checkout.components.image.loadLogo
+import com.adyen.checkout.components.status.model.TimerData
+import com.adyen.checkout.components.ui.ComponentView
+import com.adyen.checkout.components.util.CurrencyUtils
+import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.components.util.isEmpty
+import com.adyen.checkout.core.exception.PermissionException
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import com.adyen.checkout.qrcode.databinding.FullQrcodeViewBinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import java.util.concurrent.TimeUnit
+
+@Suppress("TooManyFunctions")
+internal class FullQRCodeView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr), ComponentView {
+
+    private val binding: FullQrcodeViewBinding = FullQrcodeViewBinding.inflate(LayoutInflater.from(context), this)
+
+    private lateinit var localizedContext: Context
+
+    private lateinit var delegate: QRCodeDelegate
+
+    init {
+        orientation = VERTICAL
+        val padding = resources.getDimension(R.dimen.standard_double_margin).toInt()
+        setPadding(padding, padding, padding, padding)
+    }
+
+    override fun initView(delegate: ComponentDelegate, coroutineScope: CoroutineScope, localizedContext: Context) {
+        if (delegate !is QRCodeDelegate) throw IllegalStateException("Unsupported delegate type")
+
+        this.localizedContext = localizedContext
+        initLocalizedStrings(localizedContext)
+
+        this.delegate = delegate
+
+        observeDelegate(delegate, coroutineScope)
+
+        binding.buttonSaveImage.setOnClickListener {
+            val requiredPermission = Manifest.permission.WRITE_EXTERNAL_STORAGE
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q &&
+                ContextCompat.checkSelfPermission(context, requiredPermission) != PackageManager.PERMISSION_GRANTED
+            ) {
+                delegate.onError(
+                    PermissionException(
+                        errorMessage = "storage permission is not granted",
```

Comment:
`Storage`

Reply:
or even better `Manifest.permission.WRITE_EXTERNAL_STORAGE` so that they know which one